cockatrice/src/releasechannel.cpp

```diff
@@ -219,63 +221,75 @@ QString DevReleaseChannel::getReleaseChannelUrl() const
 
-    QVariantMap resultMap = QtJson::Json::parse(tmp, ok).toMap();
-    if (!ok) {
-        qWarning() << "No reply received from the release update server:" << tmp;
+    QJsonDocument doc = QJsonDocument::fromJson(jsonData);
+    QJsonArray array = doc.array();
+
+    /*
+     * Get the latest release on GitHub
+     * This can be either a pre-release or a published release
+     * depending on timing. Both are acceptable.
+     */
+    QVariantMap resultMap = array.at(0).toObject().toVariantMap();
+
+    if (array.size() == 0 || resultMap.size() == 0) {
+        qWarning() << "No reply received from the release update server:" << QString(jsonData);
         emit error(tr("No reply received from the release update server."));
         return;
     }
 
-    if(!(resultMap.contains("commit") && 
-        resultMap.contains("html_url") && 
-        resultMap.contains("sha") &&
-        resultMap["commit"].toMap().contains("author") &&
-        resultMap["commit"].toMap()["author"].toMap().contains("date"))) {
-        qWarning() << "Invalid received from the release update server:" << tmp;
+    // Make sure resultMap has all elements we'll need
+    if (!resultMap.contains("assets") ||
+        !resultMap.contains("author") ||
+        !resultMap.contains("tag_name") ||
+        !resultMap.contains("target_commitish") ||
+        !resultMap.contains("assets_url") ||
+        !resultMap.contains("created_at"))
+    {
+        qWarning() << "Invalid received from the release update server:" << resultMap;
         emit error(tr("Invalid reply received from the release update server."));
         return;
     }
 
-    if(!lastRelease)
+    if (lastRelease == nullptr)
         lastRelease = new Release;
 
-    
-    lastRelease->setCommitHash(resultMap["sha"].toString());
-    lastRelease->setPublishDate(resultMap["commit"].toMap()["author"].toMap()["date"].toDate());
+    lastRelease->setCommitHash(resultMap["target_commitish"].toString());
+    lastRelease->setPublishDate(resultMap["created_at"].toDate());
 
     QString shortHash = lastRelease->getCommitHash().left(GIT_SHORT_HASH_LEN);
-    lastRelease->setName("Commit " + shortHash);
-
+    lastRelease->setName(QString("%1 (%2)").arg(resultMap["tag_name"].toString()).arg(shortHash));
     lastRelease->setDescriptionUrl(QString(DEVRELEASE_DESCURL).arg(VERSION_COMMIT, shortHash));
     
-    qDebug() << "Got reply from release server, size=" << tmp.size()
+    qDebug() << "Got reply from release server, size=" << resultMap.size()
         << "name=" << lastRelease->getName()
         << "desc=" << lastRelease->getDescriptionUrl()
         << "commit=" << lastRelease->getCommitHash()
         << "date=" << lastRelease->getPublishDate();
 
-    qDebug() << "Searching for a corresponding file on the dev channel: " << QString(DEVFILES_URL);
-    response = netMan->get(QNetworkRequest(QString(DEVFILES_URL)));
+    QString devbuilddownload_url = resultMap["assets_url"].toString();
```

Comment:
minor: please don't mix different variable name styles. Everything else is camelCased, this should be devBuildDownloadUrl or similar.